### ISSUE TITLE
Steam Client Maximum number of clients reached

### ISSUE THREAD

**Issue by xpander69** [ValveSoftware/steam-for-linux#9094]
#### Your system information

* Steam client version (build number or date): Jan 12 2023
* Distribution (e.g. Ubuntu): Arch Linux
* Opted into Steam client beta?: Yes
* Have you checked for system updates?: Yes

#### Please describe your issue in as much detail as possible:
Opening Steam Client beta build causes issue where after 10-30 min i cannot open any other application anymore, until i close the steam client.

[error.log](https://github.com/ValveSoftware/steam-for-linux/files/10401316/error.log)

Issue happens after the log starts spamming "Maximum number of clients reached"

#### Steps for reproducing this issue:

1. Opt into Beta
2.  Launch steam
3.  Keep it running for 10-30 min

**Comment by Leopard1907** [comment]
Same issue here, KDE/Arch Linux/Steam client beta.

https://steamcommunity.com/groups/SteamClientBeta/announcements/detail/3642882655783105850

This update triggered the issue.

**Comment by MartiUK** [comment]
Yeah, I noticed the same after the latest beta update, you can monitor the number of clients being opened by steam with `xrestop`. Steam and "web chat container" opens loads of clients which, when it hits the 256 limit, you see the "Maximum number of clients reached" error. This actually prevents you from opening ANY new application on X11/Wayland when using KDE plasma.

Removing myself from the steam beta update resolved the issue for me.

**Comment by DanielHouston** [comment]
Evident when watching vis ss as per https://unix.stackexchange.com/a/700637

**Comment by little512** [comment]
Having the same issue, I also noticed the continually rising number of clients in `xrestop`.

**Comment by nyaalex** [comment]
Same issue X11/Xorg Arch linux. Also found that a new client is created every time I activate a window, including non-steam windows. It crashes in a few seconds if I hold alt tab between two windows, for instance

**Comment by meklu** [comment]
Can confirm, had this happen on my Gentoo box. Opting out of the beta was a decent workaround for now.

**Comment by notpentadactyl** [comment]
can confirm on Arch as well (latest steam client beta). closing steam fixes the issue instantly.

**Comment by Tourfaint** [comment]
Same thing happens on ubuntu. have to opt out of beta to use steam now.

**Comment by 20PercentRendered** [comment]
Happens on Arch. KDE+Intel iGPU+Wayland

**Comment by DarkWav** [comment]
Can confirm on openSUSE Tumbleweed KDE + NVIDIA + X11.

**Comment by purplebar0** [comment]
I can confirm that this happens on my updated Arch Linux installation on today's Steam Client Beta update. Reverting fixed the issue, `steam` only makes 8 connections to X now as it should, rather than 100+.

Protip: Don't even try to raise the limit with (Xorg) `-maxclients 2048` for testing this, the same issue occurred, but it ended with freezing my entire window manager instead of just not letting me spawn more windows.

**Comment by Trezamere** [comment]
Happening on Fedora 37 as well.

**Comment by 3vi1** [comment]
Can confirm:

```
(base) 10:41:44 evil@H510 ~» sudo ss -x src "*/tmp/.X11-unix/*" | grep -Eo "[0-9]+\s*$" | while read port
do sudo ss -p -x | grep -w $port | grep -v X11-unix 
done | grep -Eo '".+"' | sort | uniq -c | sort -rn


    184 "steam"
      7 "signal-desktop-"
      5 "steamwebhelper"
      3 "xdg-desktop-por"
      3 "kglobalaccel5"
      3 "kded5"
```

**Comment by sylware** [comment]
Same issue here, xserver(xorg)/mesa(radeon/radv)/linux-amd-dev, all git from last sunday.

**Comment by dCosminn** [comment]
Same issue Ubuntu 22.04, Wayland, Mesa 22.3.3

**Comment by AmmarkoV** [comment]
Same issue with Steam Beta  and 

LxQT Ubuntu 22.04.1 LTS
Linux  5.15.0-57-generic #63-Ubuntu SMP Thu Nov 24 13:43:17 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux


sudo X -version
```
X.Org X Server 1.21.1.3
X Protocol Version 11, Revision 0
Current Operating System: Linux elina-kriti 5.15.0-57-generic #63-Ubuntu SMP Thu Nov 24 13:43:17 UTC 2022 x86_64
Kernel command line: BOOT_IMAGE=/boot/vmlinuz-5.15.0-57-generic root=UUID=fa5dc9d9-1ee5-47de-a1a8-4ab195a79b02 ro quiet splash noibrs noibpb nopti nospectre_v2 nospectre_v1 l1tf=off nospec_store_bypass_disable no_stf_barrier mds=off tsx=on tsx_async_abort=off mitigations=off vt.handoff=7
xorg-server 2:21.1.3-2ubuntu2.5 (For technical support please see http://www.ubuntu.com/support) 
Current version of pixman: 0.40.0
        Before reporting problems, check http://wiki.x.org
        to make sure that you have the latest version.
```

**Comment by 3plus4i** [comment]
yep a recent update effectively rendered Steam unusable with Linux, or more specifically, X11

https://steamcommunity.com/groups/SteamClientBeta/discussions/0/5946473955225892135/

**Comment by TTimo** [comment]
thanks, we're looking into it

**Comment by KianGreenMoon** [comment]
Gentoo, i3wm, NVIDIA - same issue. That just overflow the connection amount to X-org.

**Comment by mathew2214** [comment]
i am also having this issue. how could valve not forsee this?

**Comment by michaelBelsanti** [comment]
It's a beta, things happen. Don't use it if you want stability.

**Comment by FamiliarBreakfast** [comment]
Relavant lsof -U snippet:
https://gist.github.com/FamiliarBreakfast/c3f78511e86fd9351c9b7815f17a49b2
Taken ~15 minutes after starting steam
```bash
sudo ss -x src "*/tmp/.X11-unix/*" | grep -Eo "[0-9]+\s*$" | while read port
do sudo ss -p -x | grep -w $port | grep -v X11-unix
done | grep -Eo '".+"' | sort | uniq -c | sort -rn
    153 "steam"
    4 "steamwebhelper"
    2 "ksmserver"
    2 "ILSpy"
    1 "xsettingsd"
    1 "xembedsniproxy"
    1 "pipewire"
    1 "kded5"
    1 "kaccess"
    1 "gmenudbusmenupr"```
Running using Xwayland xorg-x11-server-Xwayland 22.1.3-2.fc37 Xorg 12201003

**Comment by NotCoolKiwi** [comment]
Same issue on Nobara (Fedora-Based)

**Comment by HurricanePootis** [comment]
Happens to me too on Arch Linux, KDE Wayland, RX 580.

**Comment by cmdrneon** [comment]
Can reproduce. Debian testing/bullseye, amdgpu (also RX 6600), Cinnamon X11.

I also thought i was going insane- apps just stopped opening. Had to use Discord's built-in text viewer to see the `.xsession-errors` log lol

**Comment by TTimo** [comment]
Fix has shipped.

**Comment by caitisgreat** [comment]
> Can reproduce. Debian testing/bullseye, amdgpu (also RX 6600), Cinnamon X11.
> 
> I also thought i was going insane- apps just stopped opening. Had to use Discord's built-in text viewer to see the `.xsession-errors` log lol

Funny thing is Cinnamon updated today too and I spent a good portion of the afternoon messing around with Timeshift to see if Cinnamon broke. 

Anyway, I’m also reporting in with this exact problem on Linux Mint 21.

**Comment by Lightsockie** [comment]
Been killing the terminal running steam and bringing it back up in a new one all day, just to see it update which made me jump here to check and low and behold the update was for this! (Have "Steam package versions: `1673573268`")

Stable so far. Using Artix Linux with a 7900 XT, ` AMD Radeon Graphics (gfx1100, LLVM 16.0.0, DRM 3.49, 6.2.0-rc2-273-tkg-cfs)`

**Comment by Tourfaint** [comment]
Downloaded the update, left steam running for like 10 minutes, and the amount of clients seem reasonable (8) and is not growing. Seems fixed. Thanks TTimo.

**Comment by isivisi** [comment]
Got the update, it only took about 10 - 20 minutes for my computer to be unusable before, been running for an hour so far with no issues :)

**Comment by parkerlreed** [comment]
Smooth sailing after the update here as well on the deck

**Comment by caitisgreat** [comment]
I’m now seeing a stable `8` X11 socket/session connections post patch. No change after 10 minutes observation.

**Comment by lucifertdark** [comment]
I've had it running for an hour so far no problems. Pretty sure this can be marked as fixed. :)

**Comment by xpander69** [comment]
The issue is fixed now indeed. Thank you.
Should i close it or?

**Comment by mbriar** [comment]
The `Maximum number of clients reached` issue is fixed for me too with the latest hotfix, however, the journal is still continuously spammed with  
```
Xlib: sequence lost (0x10000 > 0x1cfb) in reply type 0xc!
Xlib: sequence lost (0x10000 > 0x1d06) in reply type 0xc!
Xlib: sequence lost (0x10000 > 0x1d15) in reply type 0xc!
Xlib: sequence lost (0x10000 > 0x1d20) in reply type 0xc!
Xlib: sequence lost (0x10000 > 0x1d8e) in reply type 0xc!
Xlib: sequence lost (0x10000 > 0x1dae) in reply type 0xc!
Xlib: sequence lost (0x10000 > 0x1e45) in reply type 0x21!
```
while steam is running. Not sure if that's related or deserves it's own bug report.

**Comment by mferraci** [comment]
Issue is fixed for me as well, but getting the same message in journal.
` steam.desktop[97106]: Xlib: sequence lost (0x10000 > 0x6040) in reply type 0xc!
 steam.desktop[97106]: Xlib: sequence lost (0x10000 > 0x6054) in reply type 0xc!
 steam.desktop[97106]: Xlib: sequence lost (0x10000 > 0x6067) in reply type 0xc!
 steam.desktop[97106]: Xlib: sequence lost (0x10000 > 0x607d) in reply type 0xc!
 steam.desktop[97106]: Xlib: sequence lost (0x10000 > 0x6092) in reply type 0xc!
 steam.desktop[97106]: Xlib: sequence lost (0x10000 > 0x60a4) in reply type 0xc!
 steam.desktop[97106]: Xlib: sequence lost (0x10000 > 0x60b6) in reply type 0xc!
 steam.desktop[97106]: Xlib: sequence lost (0x10000 > 0x60d3) in reply type 0xc!
 steam.desktop[97106]: Xlib: sequence lost (0x10000 > 0x60e7) in reply type 0xc!
 steam.desktop[97106]: Xlib: sequence lost (0x10000 > 0x60f1) in reply type 0xc!
 steam.desktop[97106]: Xlib: sequence lost (0x10000 > 0x60f7) in reply type 0xc!
 steam.desktop[97106]: Xlib: sequence lost (0x10000 > 0x60fb) in reply type 0xc!
 steam.desktop[97106]: Xlib: sequence lost (0x10000 > 0x60ff) in reply type 0xc!
 steam.desktop[97106]: Xlib: sequence lost (0x10000 > 0x6106) in reply type 0xc!
 steam.desktop[97106]: Xlib: sequence lost (0x10000 > 0x610a) in reply type 0xc!
 steam.desktop[97106]: Xlib: sequence lost (0x10000 > 0x610e) in reply type 0xc!
 steam.desktop[97106]: Xlib: sequence lost (0x10000 > 0x6112) in reply type 0xc!
 steam.desktop[97106]: Xlib: sequence lost (0x10000 > 0x6116) in reply type 0xc!
 steam.desktop[97106]: Xlib: sequence lost (0x10000 > 0x6133) in reply type 0xc!`

Note that when steam is running in background, I don't get the messages.

**Comment by TTimo** [comment]
Please make a separate report for the xlib spam if needed. Closing this.

**Comment by grinapo** [comment]
I've got the same effect with the latest beta and wasn't able to even start a few games (namely skyrim). Had to revert to release version.

**Comment by aaronw2** [comment]
The maximum number of clients can be increased from 256.  I found this page quite helpful: [https://unix.stackexchange.com/questions/498652/does-x-window-have-a-maximum-number-limit-on-clients](url). In my case, I added the following section under /etc/X11/xorg.conf.d/10-quirks.conf:

Section "ServerFlags"
        Option "MaxClients" "2048"
EndSection

You might need to edit a different file there, or it can be added to /etc/X11/xorg.conf.

**Comment by turol** [comment]
The issue was that the handles were leaking indefinitely. Raising the limit just delays the problem and can lead to other resources running out first. And it's a moot point since it's been fixed and there's hopefully a unit test now.

**Comment by aaronw2** [comment]
> The issue was that the handles were leaking indefinitely. Raising the limit just delays the problem and can lead to other resources running out first. And it's a moot point since it's been fixed and there's hopefully a unit test now.

I just hit this the other day, so I don't think it's fixed.

**Comment by kisak-valve** [comment]
Hello @turol, @aaronw2, you're most likely looking for #9561 instead of this issue report.

**Comment by Xlaits** [comment]
Mentioning that this is still happening for me at least.

**Comment by roge** [comment]
Still happening for me as well. Xwayland refused to let any clients connect until I quit Steam.

Not absolutely sure this is related, but this happened after I got a bunch of notification spam from Steam once I woke my machine up from suspend (I guess Steam was just catching up with all the notifications it missed while the machine was asleep?). Using Ubuntu 24.04 with GNOME 46 on Wayland.

**Comment by TTimo** [comment]
Please make clean reports with full information. We don't follow up on closed issues, it's just too much noise.

**Comment by kisak-valve** [comment]
Also, before opening a new issue report for this topic, check if #9561 is covers what you are seeing.
